src/sources/generator.rs

```diff
@@ -18,25 +18,106 @@ use tokio::time::{interval, Duration};
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 #[serde(deny_unknown_fields)]
 pub struct GeneratorConfig {
-    #[serde(default)]
-    sequence: bool,
-    lines: Vec<String>,
     #[serde(default)]
     batch_interval: Option<f64>,
     #[serde(default = "usize::max_value")]
     count: usize,
+    format: OutputFormat,
+}
+
+#[derive(Clone, Debug, Derivative, Deserialize, Serialize)]
+#[derivative(Default)]
+#[serde(rename_all = "snake_case")]
+pub enum OutputFormat {
+    #[derivative(Default)]
+    RoundRobin {
+        #[serde(default)]
+        sequence: bool,
+        items: Vec<String>,
+    },
+}
+
+impl OutputFormat {
+    async fn generate(
+        &self,
+        config: &GeneratorConfig,
+        shutdown: ShutdownSignal,
+        out: Pipeline
+    ) -> Result<(), ()> {
+        match self {
+            OutputFormat::RoundRobin { sequence, items } => {
+                round_robin_generate(config, sequence, items, shutdown, out).await
+            }
+        }
+    }
+}
+
+async fn round_robin_generate(
+    config: &GeneratorConfig,
```

Comment:
Since this function takes a struct in this source as its first parameter, consider moving it (back?) into `impl GeneratorConfig` with `&self`.

Reply:
Agreed that that is a more elegant approach. Will fix.